### DISCUSSION TITLE
WebView2 Initialization Error in WinPE x64

### DISCUSSION THREAD

**Discussion by Abithk7-1** [MicrosoftEdge/WebView2Feedback#4947] [Q&A]
I'm experiencing an issue with WebView2 initialization in our WinPE x64 environment. 

The error I'm encountering is:
System.Runtime.InteropServices.COMException (0x80080005): Server execution failed (Exception from HRESULT: 0x8008005 (CO_E_SERVER_EXEC_FAILURE)) at System.Runtime.InteropServices.Marshal.ThrowExceptionForHRInternal(Int32 errorCode, IntPtr errorInfo) at Microsoft.Web.WebView2.Core.CoreWebView2Environment.d__102.MoveNext()

**Environment:**
WinPE x64
Framework: 4.8.1
I'm using the following nuget libraries:
- Microsoft.Web.WebView2.Core.dll
- Microsoft.Web.WebView2.Winforms.dll
- Microsoft.Web.WebView2.WPF.dll
The WebView2 Fixed Version Runtime binaries I'm using is:
- Microsoft.WebView2.FixedVersionRuntime.130.0.2849.56.x64

My code for initializing WebView2 is as follows:
            string exeLocation = AppDomain.CurrentDomain.BaseDirectory;
            string webView2Folder = Path.Combine(exeLocation, "WebView2x64");
            string logFilePath = Path.Combine(exeLocation, "WebView2Log.txt");
            string userDataFolder = Path.Combine(exeLocation, "UserData");

            try
            {

                // Create WebView2 environment with logging enabled
                var environment = await CoreWebView2Environment.CreateAsync(
                    browserExecutableFolder: webView2Folder,
                    userDataFolder: userDataFolder,
                    options: new CoreWebView2EnvironmentOptions("--enable-logging --v=1"));
                // Initialize WebView2 with the environment
                await webView.EnsureCoreWebView2Async(environment);
                webView.CoreWebView2.Navigate(Url);
                // Log a success message
                Log( "WebView2 initialized successfully.");
            }
            catch (Exception ex)
            {
                // Log any initialization errors
                Log($"Error initializing WebView2: {ex.ToString()}");
            }

The folder structure for the WebView2 libraries is as follows:

\<myApp>
    \Microsoft.Web.WebView2.Core.dll
    \Microsoft.Web.WebView2.Winforms.dll
    \Microsoft.Web.WebView2.WPF.dll
    \runtimes
        \win-arm64\native\WebView2Loader.dll (arm64)
        \win-x64\native\WebView2Loader.dll (x64)
        \win-x86\native\WebView2Loader.dll (x86)

**Comment by icnocop** [comment]
Is the log file created (`--enable-logging --v=1`)?
If so, what are the contents?

**Comment by T1b0uch** [comment]
Hello,
Is there any update on this matter if yes please can you help me get thought this I've tried embedding the runtime,the net,the Microsoft folder that include edge, edge core and edge WebView but without success. I don't get any error if I run either my application because the component seems like it does load correctly neither when I try to start edge on its own they just shutdown without any error.

Help please If you have any idea 💡

And thanks

**Comment by mtniehaus** [comment]
I would really like to get this to work, so let me throw in my progress:

* Windows PE runs as LocalSystem, which Edge does not like.  Add "--allow-run-as-system" to get around that,
* Windows PE is missing Media Foundation files (mf.dll, mfplat.dll), copy those from a Windows install into X:\windows\system32.
* Windows PE doesn't have a user profile folder.  Specify something like --user-data-dir="x:\temp" (needs to exist, create it first).

I've experimented with a variety of other options, but they all seem to fail.  "--headless" gets it to run, but without a UI, so it seems very likely that this is GPU-related.

I've attached a log from my most recent attempt.

[logfile.txt](https://github.com/user-attachments/files/21875800/logfile.txt)
